Backend/src/test/java/com/example/Backend/services/RegisterServiceTest.java

```diff
@@ -0,0 +1,91 @@
+package com.example.Backend.services;
+
+import com.example.Backend.entity.UserContactDetails;
+import com.example.Backend.entity.UserRegisterDetails;
+import com.example.Backend.exceptions.InvalidInputException;
+import com.example.Backend.model.UserRegister;
+import com.example.Backend.respository.UserRepository;
+import com.example.Backend.validator.UserRegisterValidator;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.dao.DataIntegrityViolationException;
+
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+@Slf4j
+@DataJpaTest
+class RegisterServiceTest {
+
+    @Autowired
+    UserRepository testUserRepository;
+    RegisterServiceInterface registerServiceInterface;
```

Comment:
There should not be Interface word in the name itself. That's why then class which we implement this interface will be named as RegisterServiceImpl and the interface name should be RegisterService